src/main/java/com/eppo/sdk/helpers/ExperimentConfigurationRequestor.java

```diff
@@ -34,35 +38,17 @@ public Optional<ExperimentConfigurationResponse> fetchExperimentConfiguration()
                 ObjectMapper objectMapper = new ObjectMapper();
                 config = objectMapper.readValue(response.body(), ExperimentConfigurationResponse.class);
             }
-
-            // Set if next request is allowed or not
-            this.setIsRequestAllowed(statusCode);
-        } catch (Exception e) {
+            if (statusCode == 401) { // unauthorized - invalid API key
+                throw new InvalidApiKeyException("Unauthorized: invalid Eppo API key.");
+            }
```

Comment:
Are you always going to be sure this is the only reason for a 401?

Reply:
I can't think of other scenarios for a 401.